src/ts/signers.ts

```diff
@@ -0,0 +1,127 @@
+import { _TypedDataEncoder } from "@ethersproject/hash";
+import type { JsonRpcProvider } from "@ethersproject/providers";
+import { ethers, Signer } from "ethers";
+
+import {
+  TypedDataDomain,
+  TypedDataSigner,
+  TypedDataTypes,
+} from "./types/ethers";
+
+/**
+ * Wrapper around a TypedDataSigner Signer object that implements `_signTypedData` using
+ * `eth_signTypedData_v3` instead of `eth_signTypedData_v4`.
+ *
+ * Takes a Signer instance on creation.
+ * All other Signer methods are proxied to initial instance.
+ */
+export class TypedDataV3Signer implements TypedDataSigner {
+  signer: Signer;
+  provider?: JsonRpcProvider | undefined;
+  _isSigner = true;
+
+  constructor(signer: Signer) {
+    this.signer = signer;
+    this.provider = signer.provider as JsonRpcProvider;
```

Comment:
Here we could throw if `signer.provider` is sot set to catch a failure before trying to sign. We could also check if `signer.provider.send` is defined to make the type casting safer.

Reply:
THis is a nice idea. We can even add a function in `types/ethers.ts`:
```ts
export function isJsonRpcProvider(provider: Provider): provider is JsonRpcProvider {    
  return "send" in provider;
}
```